pkg/sdn/plugin/controller.go

```diff
@@ -458,7 +462,10 @@ func (plugin *OsdnNode) AddServiceRules(service *kapi.Service, netID uint32) {
 
 	otx := plugin.ovs.NewTransaction()
 	for _, port := range service.Spec.Ports {
+		// Add one flow matching port #s for unfragmented packets and the first fragment
 		otx.AddFlow(generateAddServiceRule(netID, service.Spec.ClusterIP, port.Protocol, int(port.Port)))
+		// Add another flow matching port 0 to allow subsequent fragments through
+		otx.AddFlow(generateAddServiceRule(netID, service.Spec.ClusterIP, port.Protocol, int(0)))
```

Comment:
There's no need to have per-service-IP flows for this from a security or correctness point of view. (And in your case where the user intentionally creates two services with the same IP, this would end up deleting the port 0 rule for the IP after *any* of the services with that IP was deleted.)